ddtrace/tracer/stats.go

```diff
@@ -54,17 +54,18 @@ type concentrator struct {
 	// stopped reports whether the concentrator is stopped (when non-zero)
 	stopped uint64
 
-	wg         sync.WaitGroup // waits for any active goroutines
-	bucketSize int64          // the size of a bucket in nanoseconds
-	stop       chan struct{}  // closing this channel triggers shutdown
-	cfg        *config        // tracer startup configuration
+	wg         *sync.WaitGroup // waits for any active goroutines
```

Comment:
Please don't make a pointer out of this. Waitgroups are generally not declared as such

Reply:
Interesting, I've always used the opposite approach to make accidentally copying the waitgroup more difficult, what's the reasoning behind keeping it as not a pointer?

Reply:
I think the question goes the other way: what's the reason to have it as a pointer? You never have to initialise it, it's ready-to-use. How often does it happen that you need to pass it around? And if you do need to pass it around, it can always be dereferenced.

I frankly have never seen it passed as a pointer in official Go projects (e.g. [the lang](https://cs.github.com/?scope=org%3Agolang&scopeName=golang&q=sync.waitgroup))

Reply:
My thought process was that all the methods on wait group have a pointer receiver so using it as a pointer makes it slightly more obvious that's the typical way to use it. In this specific spot I changed it to a pointer as part of debugging to make sure we weren't accidentally copying it since the tests were failing in an odd way. I'll switch this back to non-pointer since it's not really bringing a ton of value (and if we do accidentally copy it the unit tests will protect us)

Reply:
Please ping me if you do discover that having it as a pointer provides some kind of benefit or that not having it has caused some sort of issue. Currently, it just seems a lot easier to use because it's ready-to-use without the need for initialising.